### ISSUE TITLE
link reference holds wrong URL

### ISSUE THREAD

**Issue by heinrichmartin** [roundcube/roundcubemail#8021]
My provider has Roundcube Webmail 1.4.11. Roundcube or one of the plugins renders references to links as link, i.e. in `see [1]` the number would become clickable with the URL that is given near the end of the email `[1] https://example.com`.

Given a plain-text email with such reference(s) was replied with more text and more references (i.e. the reply contains `see my link [1]`), then the `href` of this number holds the URL of the quoted email.

In the example below, both references `[1]` point to https://example.com. The best fix would find references per quoted email (level of quotation), but imho, it is acceptable to just get the actual email right and fully ignore the quoted one(s).

Content-Type: text/plain; charset=UTF-8; format=flowed
> Testing the new link [1]
> [1] https://example.org
>
> you wrote:
> > My link [1]
> > [1] https://example.com

**Comment by alecpl** [comment]
I'm seeing kinda different issue. Could you provide precise steps to reproduce?

**Comment by heinrichmartin** [comment]
> Could you provide precise steps to reproduce?

1. Compose a plain text message to yourself:

```
Testing the new link [1]

[1] https://github.com/roundcube/roundcubemail/issues/8021

you wrote:
> My link [1]
>
> [1] https://example.com
```

2. View the message in your inbox

Expected behavior: ``1`` in the first line renders as a link to https://github.com/roundcube/roundcubemail/issues/8021

Actual behavior: ``1`` in the first line renders as a link to https://example.com

**Comment by alecpl** [comment]
All right. That's one thing. Thanks. The other thing might be how to prevent that happening when such a content is being created, if it was created by Roundcube.

**Comment by heinrichmartin** [comment]
> The other thing might be how to prevent that happening when such a content is being created, if it was created by Roundcube.

Honestly, I created this issue with my user experience in mind (without research how markdown renderers deal with this situation). I am fully aware that others might argue that re-using reference ids is illegal or that one might want to re-use a reference from the quoted text, but I prefer the other use case:
- the author of the new text need not care about ids in quoted text
- the renderer shall
  - look for references only in the same level of quotation or
  - ignore all references in quoted text

I do _not_ see an issue in the editor (but I haven't played with it).

**Comment by alecpl** [comment]
Fixed.

**Comment by alecpl** [comment]
The fix didn't work on PHP < 7.4. Reverted. I'll look for a different solution.

**Comment by alecpl** [comment]
Fixed.
